jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -386,15 +310,28 @@ private MinimumDataRateRequest(Request request)
                 super(request);
             }
 
+            private long dataRatePerSecond(long dataCount)
+            {
+                if (dataCount == 0L)
+                    return 0L;
+                long delayInNs = NanoTime.since(getNanoTime());
+                // If you read 1 byte or more in 0ns or less, you have infinite bandwidth.
```

Comment:
Not really infinite, just 1 GiB/s, which may be possible via `localhost` or even fast networks.

I think the calculation should be: `dataCount * 1_000_000_000 / delayInNs`, no `float`s, no casts.

Reply:
If `delayInNs` is 0 and `dataCount` is greater than 0, you've sent some bytes in less than 1ns which means either you went back through time or your nanotime clock is ticking backwards. In both cases, it's impossible to accurately measure a data rate so I assume infinite bandwidth.

I've replaced the floating point casting and math with your proposed 64 bit integer math as it's always nice to leave the FPU registers free for the JIT to perform loop vectorization.